### ISSUE TITLE
Panic when a synonym is linked to a separator

### ISSUE THREAD

**Issue by ManyTheFish** [meilisearch/meilisearch#3977]
**Describe the bug**
When a synonym is set between a word and a token considered a separator, like 'and' => ['&'], a panic occurs when searching the word 

**To Reproduce**
1. add a synonym `'and' => ['&']`
2. search for the word `and`
3. it panics

**Expected behavior**
No panic

**Meilisearch version:**
v1.3.0

**Additional context**
reporter: @brunoocasali

**Comment by ManyTheFish** [comment]
Hello @brunoocasali and @francoism90,

after some tries, I don't manage to reproduce the bug on my side, I set some synonyms:
```json
{
  "synonyms": {"&": ["to"], "to": ["&"]}
}
```

then I searched for:
- "How to train"
- "How & train"
- "to"

But none of these queries triggered a panic on Meilisearch.
Did I missed something 🤔 

Thank you for your help!

**Comment by francoism90** [comment]
@ManyTheFish Could you try `and` instead? :)

**Comment by curquiza** [comment]
Hello @francoism90 and @ManyTheFish 
I tried with `and` and I did not get any crash

1. Documents
```json
[
  { "id": 1,    "title": "1 & 2"},
  { "id": 2,    "title": "1 and 3"}
]
```

2. Synonyms
```json
{"&": ["and"], "and": ["&"]}
```

3. Searches
- `&` returns both documents
- `and` return only the second document -> is it a bug @ManyTheFish?
-> but no crash


Can you send us the exact steps you do to get the crash @francoism90?

**Comment by curquiza** [comment]
@francoism90
- can you show the index settings you put?
- can you send your dataset, or a part of your dataset that makes things crash?
- when you say "Weird it didn't result in any crash in older versions" -> which versions are you talking about? Do you use v1.3.0?

**Comment by francoism90** [comment]
@ManyTheFish Thanks, I'll try later today to check if it's fixed. :)

**Comment by curquiza** [comment]
We need to release v1.3.1 today, so I remove this issue from the Milestones
Looking forward to have your feedback @francoism90 😄

**Comment by francoism90** [comment]
@curquiza @ManyTheFish This indeed seems to fix the issue! :)

I've used the following synonyms:
```php
'&' => ['and'],
'and' => ['&'],
```
It doesn't crash on `and`  queries anymore.

Thanks!

**Comment by curquiza** [comment]
Thank you very much @francoism90 
We will not be able to release this fix for v1.3.1 because we just launched the release 😅 
We will add it to v1.3.2 or v1.4.0

**Comment by curquiza** [comment]
Closed by #3994
